types/src/state_store/state_value.rs

```diff
@@ -186,27 +168,23 @@ enum PersistedStateValue {
 }
 
 impl PersistedStateValue {
-    fn into_in_mem_form(self) -> StateValueInner {
+    fn into_in_mem_form(self) -> StateValue {
         match self {
-            PersistedStateValue::V0(data) => StateValueInner {
-                data,
-                metadata: StateValueMetadata::none(),
-            },
-            PersistedStateValue::WithMetadata { data, metadata } => StateValueInner {
-                data,
-                metadata: metadata.into_in_mem_form(),
+            PersistedStateValue::V0(data) => StateValue::new_legacy(data),
+            PersistedStateValue::WithMetadata { data, metadata } => {
+                StateValue::new_with_metadata(data, metadata.into_in_mem_form())
             },
         }
     }
 }
 
-#[derive(Clone, Debug, Eq, PartialEq)]
-struct StateValueInner {
+#[derive(BCSCryptoHash, Clone, CryptoHasher, Debug, Eq, PartialEq)]
+pub struct StateValue {
```

Comment:
With these changes, does the state value hash to the same thing? Before it was derived from `to_persistable_form`